src/main/java/com/zendesk/maxwell/producer/AbstractProducer.java

```diff
@@ -1,18 +1,33 @@
 package com.zendesk.maxwell.producer;
 
-import com.codahale.metrics.Meter;
+import com.codahale.metrics.*;
 import com.zendesk.maxwell.MaxwellContext;
 import com.zendesk.maxwell.monitoring.MaxwellDiagnostic;
+import com.zendesk.maxwell.monitoring.Metrics;
 import com.zendesk.maxwell.row.RowMap;
 import com.zendesk.maxwell.util.StoppableTask;
 
 public abstract class AbstractProducer {
 	protected final MaxwellContext context;
 	protected final MaxwellOutputConfig outputConfig;
+	protected final Counter succeededMessageCount;
+	protected final Meter succeededMessageMeter;
+	protected final Counter failedMessageCount;
+	protected final Meter failedMessageMeter;
+	protected final Timer metricsTimer;
```

Comment:
Since we have these metrics in base class, can we clean up those produces who use the metrics?

Reply:
> clean up those producers who use the metrics

Sorry, I don't understand what you mean.

Reply:
Can remove these fields: https://github.com/zendesk/maxwell/blob/master/src/main/java/com/zendesk/maxwell/producer/AbstractAsyncProducer.java#L13-L17

Reply:
and this: https://github.com/zendesk/maxwell/blob/master/src/main/java/com/zendesk/maxwell/producer/AbstractAsyncProducer.java#L13-L17

Reply:
Ooohh. Gotcha. I didn't even notice that AbstractAsyncProducer extends AbstractProducer. 😁

Reply:
done 52e6fcbb

also moved `getFailedMessageMeter` to the AbstractProducer class